src/DataProtection/DataProtection/src/XmlEncryption/XmlEncryptionExtensions.cs

```diff
@@ -95,6 +97,26 @@ private static IXmlDecryptor CreateDecryptor(IActivator activator, string decryp
         return activator.CreateInstance<IXmlDecryptor>(decryptorTypeName);
     }
 
+    private static bool TryGetDecryptorType(string decryptorTypeName, [NotNullWhen(true)] out Type? type)
+    {
+        var resolvedTypeName = TypeForwardingActivator.TryForwardTypeName(decryptorTypeName, out var forwardedTypeName)
+            ? forwardedTypeName
+            : decryptorTypeName;
+        try
+        {
+            type = _getType(resolvedTypeName);
```

Comment:
Nit: I like the comment in your PR description about how this can still throw when `throwIfFalse` is set.  Can you please put something like that here?

Reply:
Done!